src/cluster.h

```diff
@@ -173,257 +53,83 @@ typedef struct {
-static_assert(offsetof(clusterMsg, flags) == 2250, "unexpected field offset");
-static_assert(offsetof(clusterMsg, state) == 2252, "unexpected field offset");
-static_assert(offsetof(clusterMsg, mflags) == 2253, "unexpected field offset");
-static_assert(offsetof(clusterMsg, data) == 2256, "unexpected field offset");
-
-#define CLUSTERMSG_MIN_LEN (sizeof(clusterMsg)-sizeof(union clusterMsgData))
-
-/* Message flags better specify the packet content or are used to
- * provide some information about the node state. */
-#define CLUSTERMSG_FLAG0_PAUSED (1<<0) /* Master paused for manual failover. */
-#define CLUSTERMSG_FLAG0_FORCEACK (1<<1) /* Give ACK to AUTH_REQUEST even if
-                                            master is up. */
-#define CLUSTERMSG_FLAG0_EXT_DATA (1<<2) /* Message contains extension data */
-
 /* ---------------------- API exported outside cluster.c -------------------- */
+/*
+ * Functions requiring per-clustering mechanism implementation: Lifecycle events
+ */
 void clusterInit(void);
-void clusterInitListeners(void);
+void clusterInitLast(void);
 void clusterCron(void);
 void clusterBeforeSleep(void);
-clusterNode *getNodeByQuery(client *c, struct redisCommand *cmd, robj **argv, int argc, int *hashslot, int *ask);
-int verifyClusterNodeId(const char *name, int length);
-clusterNode *clusterLookupNode(const char *name, int length);
-int clusterRedirectBlockedClientIfNeeded(client *c);
-void clusterRedirectClient(client *c, clusterNode *n, int hashslot, int error_code);
-void migrateCloseTimedoutSockets(void);
 int verifyClusterConfigWithData(void);
+void clusterUpdateMyselfFlags(void);
 unsigned long getClusterConnectionsCount(void);
+void clusterUpdateMyselfHostname(void);
+void clusterUpdateMyselfAnnouncedPorts(void);
+void clusterUpdateMyselfIp(void);
+
+/*
+ * Functions requiring per-clustering mechanism implementation
+ */
+void clusterCommand(client *c);
+int clusterCommandSpecial(client *c);
+const char** clusterCommandSpecialHelp(void);
 int clusterSendModuleMessageToTarget(const char *target, uint64_t module_id, uint8_t type, const char *payload, uint32_t len);
 void clusterPropagatePublish(robj *channel, robj *message, int sharded);
+void slotToChannelAdd(sds channel);
+void slotToChannelDel(sds channel);
+int handleDebugClusterCommand(client *c);
+sds clusterGenNodesDescription(client *c, int filter, int use_pport);
+sds genClusterInfoString(void);
+char* clusterNodeIp(clusterNodeHandle node);
+int clusterNodePort(clusterNodeHandle node);
+clusterNodeHandle clusterNodeGetSlaveof(clusterNodeHandle node);
+int clusterNodeConfirmedReachable(clusterNodeHandle node);
+int clusterNodeIsMaster(clusterNodeHandle node);
+int clusterNodeIsSlave(clusterNodeHandle node);
+int clusterNodeIsFailing(clusterNodeHandle node);
+int clusterNodeTimedOut(clusterNodeHandle node);
+int clusterNodeIsMyself(clusterNodeHandle node);
+int clusterNodeIsNoFailover(clusterNodeHandle node);
+char* clusterNodeGetName(clusterNodeHandle node);
```

Comment:
```suggestion
char *clusterNodeGetName(clusterNodeHandle node);
```
Minor style point. Let's be consistent.